src/extension/index.ts

```diff
@@ -58,17 +58,19 @@ export async function activate(context: ExtensionContext) {
         async handleUri(uri: vscode.Uri) {
             if (uri.path === '/alert') {
                 // Launched by Azure DevOps Advanced Security alert page.
-                const pairs = uri.query.split('&');
+                const params = new URLSearchParams(uri.query);
+
                 // Find the url parameter.
-                for (const param in pairs) {
-                    const pair = param.split('=');
-                    if (pair[0] === 'url') {
+                for (const param of params) {
+                    if (param[0] === 'url') {
                         // Decode the alert API URL and pass it to the load function.
-                        const url = decodeURIComponent(pair[1]);
+                        const url = decodeURIComponent(param[1]);
                         if (url.startsWith('https://advsec.dev.azure.com/')) {
                             await loadAlertSarif(new URL(url));
-                            break;
+                        } else {
+                            await window.showWarningMessage(`Invalid callback URL '${url}'. URL must point to 'https://advsec.dev.azure.com/'.`, 'OK');
```

Comment:
There needs to be a `throw` or `return` here, otherwise the call to `fetch(url)` will happen and the access token will be sent to an arbitrary url. 

Also, probably you want to change `await` to `void` since we most likely don't want to stop execution while waiting for the user to close the dialog.

Reply:
Not sure on the first point. If the param is named 'url', it breaks out of the for loop and that's the end of the function.

Reply:
I'm sorry. You're right. The function finishes. Please ignore.